pkg/syncer/cnsoperator/controller/csinodetopology/csinodetopology_controller.go

```diff
@@ -457,15 +465,26 @@ func updateCRStatus(ctx context.Context, r *ReconcileCSINodeTopology, instance *
 	return nil
 }
 
-func getNodeTopologyInfo(ctx context.Context, nodeVM *cnsvsphere.VirtualMachine, cfg *cnsconfig.Config) (
-	[]csinodetopologyv1alpha1.TopologyLabel, error) {
+func getNodeTopologyInfo(ctx context.Context, nodeVM *cnsvsphere.VirtualMachine, cfg *cnsconfig.Config,
+	isMultiVCFSSEnabled bool) ([]csinodetopologyv1alpha1.TopologyLabel, error) {
 	log := logger.GetLogger(ctx)
-
+	var (
+		vcenter *cnsvsphere.VirtualCenter
+		err     error
+	)
 	// Get VC instance.
-	vcenter, err := cnsvsphere.GetVirtualCenterInstance(ctx, &cnsconfig.ConfigurationInfo{Cfg: cfg}, false)
-	if err != nil {
-		log.Errorf("failed to get virtual center instance with error: %v", err)
-		return nil, err
+	if isMultiVCFSSEnabled {
+		vcenter, err = cnsvsphere.GetVirtualCenterInstanceForVCenterHost(ctx, nodeVM.VirtualCenterHost)
+		if err != nil {
+			return nil, logger.LogNewErrorf(log, "failed to get vCenterInstance for vCenter Host: %q, err: %v",
+				nodeVM.VirtualCenterHost, err)
+		}
+	} else {
+		vcenter, err = cnsvsphere.GetVirtualCenterInstance(ctx, &cnsconfig.ConfigurationInfo{Cfg: cfg}, false)
+		if err != nil {
+			log.Errorf("failed to get virtual center instance with error: %v", err)
```

Comment:
use `logger.LogNewErrorf`

Reply:
This is the old code, do you want me to change that as well?

Reply:
In that case, it is fine.